sdk/keyvault/keyvault/src/secrets/secretsModels.ts

```diff
@@ -0,0 +1,197 @@
+import * as msRest from "@azure/ms-rest-js";
+
+export interface Secret extends SecretAttributes {
+  /**
+   * @member {string} [value] The secret value.
+   */
+  value?: string;
+}
+
+export interface SecretAttributes extends ParsedKeyVaultEntityIdentifier {
+  /**
+   * @member {string} [id] The secret id.
+   */
+  id?: string;
+  /**
+   * @member {string} [contentType] The content type of the secret.
+   */
+  contentType?: string;
+  /**
+   * @member {boolean} [enabled] Determines whether the object is enabled.
+   */
+  enabled?: boolean;
+  /**
+   * @member {Date} [notBefore] Not before date in UTC.
+   */
+  notBefore?: Date;
+  /**
+   * @member {Date} [expires] Expiry date in UTC.
+   */
+  expires?: Date;
+  /**
+   * @member {{ [propertyName: string]: string }} [tags] Application specific
+   * metadata in the form of key-value pairs.
+   */
+  tags?: { [propertyName: string]: string };
+  /**
+   * @member {string} [keyId] If this is a secret backing a KV certificate, then
+   * this field specifies the corresponding key backing the KV certificate.
+   * **NOTE: This property will not be serialized. It can only be populated by
+   * the server.**
+   */
+  readonly keyId?: string;
+  /**
+   * @member {boolean} [managed] True if the secret's lifetime is managed by
+   * key vault. If this is a secret backing a certificate, then managed will be
+   * true.
+   * **NOTE: This property will not be serialized. It can only be populated by
+   * the server.**
+   */
+  readonly managed?: boolean;
+}

```

Comment:
seems to be missing readonly properties created, updated and recoveryLevel

Reply:
Will do. I'll update the PR.